pkg/apply/desiredset_process.go

```diff
@@ -433,3 +433,32 @@ func addObjectToMap(objs map[objectset.ObjectKey]runtime.Object, obj interface{}
 
 	return nil
 }
+
+// multiNamespaceList lists objects across all given namespaces, because requests are concurrent it is possible for appendFn to be called before errors are reported.
+func multiNamespaceList(ctx context.Context, namespaces []string, baseClient dynamic.NamespaceableResourceInterface, selector labels.Selector, appendFn func(obj unstructured.Unstructured)) error {
+	var mu sync.Mutex
+	wg, _ctx := errgroup.WithContext(ctx)
+
+	// list all namespaces concurrently
+	for _, namespace := range namespaces {
+		namespace := namespace
```

Comment:
I understand we create a copy of namespace but this statement still looks weird to me. In previous code we've had `copy := obj` in the similar situation so maybe we could do `namespaceCopy := namespace` for clarity.
If this `a := a` construct is something we regularly use I'm OK with keeping it as is.

Reply:
declaring it with the same name has the same effect as `copy := obj`.  I have seen this pattern elsewhere in rancher but I don't think it is standardized.  I would argue this is a safer method,  it prevents someone from absentmindedly referencing `namespace` instead of `namespaceCopy`.  That mistake can sometimes hide and produces really strange behavior that can be hard to debug later.

Reply:
I agree with your rationale here. Let's keep it as is, thank you.